### ISSUE TITLE
Feature Request: clarity on AppMesh or Envoy Image in ECS

### ISSUE THREAD

**Issue by colinbjohnson** [aws/aws-app-mesh-roadmap#306]
*If you want to see App Mesh implement this idea, please upvote with a :+1:.*

**Tell us about your request**

As of 2021-01-18 the documentation for ECS Appmesh advises the use of an `aws-appmesh-envoy:v1.16.1.0-prod` image and references that the official Envoy image can be used once https://github.com/aws/aws-app-mesh-roadmap/issues/10 is closed.

This issue has been closed, so it is now unclear which image should be used and why.

**Which integration(s) is this request for?**

AWS ECS, possibly others.

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.

**Are you currently working around this issue?**

Currently I'm using the `aws-appmesh-envoy:v1.16.1.0-prod` image, but it is not clear which image I should be using and why.

**Additional context**
Anything else we should know?

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by dastbe** [comment]
Hey @colinbjohnson!

the tl;dr is that the recommended thing to do is use the `aws-appmesh-envoy` image. It not only has been vetted by us to work with App Mesh (we qualify every release with validation tests that exercise a bunch of features and scenarios) but it also includes a bootstrap executable that simplifies setting up Envoy configuration.

Shipping #10 meant that, if you would like to, you can build your own Envoy from upstream and connect that to App Mesh. This is for customers who have stricter requirements around the provenance of their Envoy image, or have changes they'd like to make to the source. Customers who go this route will have to handle a fair bit of what our bootstrap wrapper does, and we've (unfortunately) not well documented the expectations.

In the fullness of time (tm), we'd like to open source all of our build tooling for Envoy so that its straightforward to build an image of Envoy that also is straightforward to use with App Mesh, but that hasn't been a priority for us.

**Comment by colinbjohnson** [comment]
@dastbe - you are awesome for replying to this - I can take some responsibility here for creating a PR for the docs site - I'll include you on the PR?
